### PR TITLE
Refactor Sqlalchemy queries to 2.0 style (Part 4)

### PR DESCRIPTION
**Pull request by phanikumv**
This is a continuation of the effort to refactor the queries to sqlalchemy 2.0

related: https://github.com/apache/airflow/issues/28723

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
applied "full tests needed" to that one as well.

File: tests/cli/commands/test_connection_command.py [hunk 1]
```diff
@@ -167,9 +168,7 @@ def test_cli_connections_export_should_raise_error_if_fetching_connections_fails
         def my_side_effect(_):
             raise Exception("dummy exception")
 
-        mock_session.return_value.__enter__.return_value.query.return_value.order_by.side_effect = (
-            my_side_effect
-        )
+        mock_session.return_value.__enter__.return_value.scalars.side_effect = my_side_effect
```
Comment (ephraimbuddy) [inline tests/cli/commands/test_connection_command.py:170]: Apart from this change, I think we shouldn't change any other test's query, that way, we will quickly know if there's an issue with our change

Comment (phanikumv) [reply tests/cli/commands/test_connection_command.py:170]: @ephraimbuddy so you mean we should retain `session.query` everywhere else in the test?

Comment (phanikumv) [reply tests/cli/commands/test_connection_command.py:170]: Done in [66d79eb](https://github.com/apache/airflow/pull/32339/commits/66d79ebd1af7b79b70bca7ce4bb07cc718dd49ed)

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**